exporter/clients.go

```diff
@@ -103,30 +181,185 @@ func (e *Exporter) extractConnectedClientMetrics(ch chan<- prometheus.Metric, c
 	}
 
 	for _, c := range strings.Split(reply, "\n") {
-		if lbls, ok := parseClientListString(c); ok {
-			connectedClientsLabels := []string{"name", "created_at", "idle_since", "flags", "db", "omem", "cmd", "host"}
-			connectedClientsLabelsValues := []string{lbls.Name, lbls.CreatedAt, lbls.IdleSince, lbls.Flags, lbls.Db, lbls.OMem, lbls.Cmd, lbls.Host}
+		if info, ok := parseClientListString(c); ok {
+			clientInfoLabels := []string{"id", "name", "flags", "db", "host"}
+			clientInfoLabelsValues := []string{info.Id, info.Name, info.Flags, info.Db, info.Host}
 
 			if e.options.ExportClientsInclPort {
-				connectedClientsLabels = append(connectedClientsLabels, "port")
-				connectedClientsLabelsValues = append(connectedClientsLabelsValues, lbls.Port)
+				clientInfoLabels = append(clientInfoLabels, "port")
+				clientInfoLabelsValues = append(clientInfoLabelsValues, info.Port)
 			}
 
-			if user := lbls.User; user != "" {
-				connectedClientsLabels = append(connectedClientsLabels, "user")
-				connectedClientsLabelsValues = append(connectedClientsLabelsValues, user)
+			if user := info.User; user != "" {
+				clientInfoLabels = append(clientInfoLabels, "user")
+				clientInfoLabelsValues = append(clientInfoLabelsValues, user)
 			}
 
-			if resp := lbls.Resp; resp != "" {
-				connectedClientsLabels = append(connectedClientsLabels, "resp")
-				connectedClientsLabelsValues = append(connectedClientsLabelsValues, resp)
+			// introduced in Redis 7.0
+			if resp := info.Resp; resp != "" {
+				clientInfoLabels = append(clientInfoLabels, "resp")
+				clientInfoLabelsValues = append(clientInfoLabelsValues, resp)
 			}
 
-			e.metricDescriptions["connected_clients_details"] = newMetricDescr(e.options.Namespace, "connected_clients_details", "Details about connected clients", connectedClientsLabels)
+			e.metricDescriptions["connected_client_info"] = newMetricDescr(
+				e.options.Namespace,
+				"connected_client_info",
+				"Details about a connected client",
+				clientInfoLabels,
+			)
+			e.registerConstMetricGauge(
+				ch, "connected_client_info", 1.0,
+				clientInfoLabelsValues...,
+			)
 
+			// keep the old name for backwards compatability
```

Comment:
I know I said to keep the old metrics for backwards compatibility but seeing this now makes me reconsider that. It's not going to be fully backwards compatible anyway (labels will have changed) so let's just toss the old metric and make this cleaner.